### PR TITLE
[zh-cn] add blog/2022-09-02-cosi-kubernetes-object-storage-management.md

### PR DESCRIPTION
**Pull request by my-git9**
[zh-cn] add blog/2022-09-02-cosi-kubernetes-object-storage-management.md

### CODE REVIEW

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
+---
+layout: blog
+title: "COSI 简介：使用 Kubernetes API 的对象存储管理"
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:3]:
```suggestion
title: "COSI 简介：使用 Kubernetes API 管理对象存储"
```

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+近年来，对象存储作为文件系统和块设备的替代存储形式越来越受欢迎。
+对象存储范式促进了计算和存储的分解,，这是通过网络而不是本地提供数据来完成的。
+分解架构允许计算工作负载是无状态的，从而使它们更易于管理、扩展和自动化。
+
+## COSI
+
+<!--
+## COSI
+
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:43]:
```suggestion
```

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+**作者:** Sidhartha Mani ([Minio, Inc](https://min.io))
+
+<!--
+This article introduces the Container Object Storage Interface (COSI), a standard for provisioning and consuming object storage in Kubernetes. It is an alpha feature in Kubernetes v1.25.
+
+File and block storage are treated as first class citizens in the Kubernetes ecosystem via [Container Storage Interface](https://kubernetes.io/blog/2019/01/15/container-storage-interface-ga/) (CSI). Workloads using CSI volumes enjoy the benefits of portability across vendors and across Kubernetes clusters without the need to change application manifests. An equivalent standard does not exist for Object storage.
+
+Object storage has been rising in popularity in recent years as an alternative form of storage to filesystems and block devices. Object storage paradigm promotes disaggregation of compute and storage. This is done by making data available over the network, rather than locally. Disaggregated architectures allow compute workloads to be stateless, which consequently makes them easier to manage, scale and automate.
+-->
+本文介绍了容器对象存储接口 (COSI)，它是在 Kubernetes 中供给和使用对象存储的标准。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:27]:
```suggestion
本文介绍了容器对象存储接口 (COSI)，这是在 Kubernetes 中制备和消耗对象存储的一个标准。
```

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+它是 Kubernetes v1.25 中的一个 alpha 功能。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:28]:
```suggestion
它是 Kubernetes v1.25 中的一个 Alpha 特性。
```

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+文件和块存储通过 [Container Storage Interface](https://kubernetes.io/blog/2019/01/15/container-storage-interface-ga/) (CSI)
+被视为 Kubernetes 生态系统中的一等公民。
+使用 CSI 卷的工作负载可以享受跨供应商和跨 Kubernetes 集群的可移植性优势，
+而无需更改应用程序清单。对象存储不存在等效标准。
+
+近年来，对象存储作为文件系统和块设备的替代存储形式越来越受欢迎。
+对象存储范式促进了计算和存储的分解,，这是通过网络而不是本地提供数据来完成的。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:36]:
```suggestion
对象存储范式促进了计算和存储的分解，这是通过网络而不是本地提供数据来完成的。
```

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+分解架构允许计算工作负载是无状态的，从而使它们更易于管理、扩展和自动化。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:37]:
```suggestion
分解的架构允许计算工作负载是无状态的，从而使它们更易于管理、扩缩和自动化。
```

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+* Kubernetes Native - Use the Kubernetes API to provision, configure and manage buckets
+* Self Service - A clear delineation between administration and operations (DevOps) to enable self-service capability for DevOps personnel
+* Portability - Vendor neutrality enabled through portability across Kubernetes Clusters and across Object Storage vendors
+
+_Portability across vendors is only possible when both vendors support a common datapath-API. Eg. it is possible to port from AWS S3 to Ceph, or AWS S3 to MinIO and back as they all use S3 API. In contrast, it is not possible to port from AWS S3 and Google Cloud’s GCS or vice versa._
+-->
+COSI 旨在标准化对象存储的使用，以提供以下好处：
+
+* Kubernetes 原生 - 使用 Kubernetes API 来配置、配置和管理 Bucket
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:54]:
```suggestion
* Kubernetes 原生 - 使用 Kubernetes API 来制备、配置和管理 Bucket
```

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+* 自助服务 - 管理和运营 (DevOps) 之间的明确划分，为 DevOps 人员启用自助服务能力
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:55]:
```suggestion
* 自助服务 - 明确划分管理和运营 (DevOps)，为 DevOps 人员赋予自助服务能力
```

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+* 可移植性——通过跨 Kubernetes 集群和跨对象存储供应商的可移植性实现供应商中立性
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:56]:
```suggestion
* 可移植性 - 通过跨 Kubernetes 集群和跨对象存储供应商的可移植性实现供应商中立性
```

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+
+**跨供应商的可移植性只有在两家供应商都支持通用数据路径 API 时才有可能。
+例如，可以从 AWS S3 移植到 Ceph，或从 AWS S3 移植到 MinIO 并返回，因为它们都使用 S3 API。
```
Comment (windsonsea) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:59]:
```suggestion
例如，可以从 AWS S3 移植到 Ceph，或从 AWS S3 移植到 MinIO 以及反向操作，因为它们都使用 S3 API。
```

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+## 可移植性
+COSI 的第三个目标是实现 Bucket 管理的供应商中立性。 COSI 支持两种可移植性：
+
+* 跨集群
+* 跨提供商
+
+跨集群可移植性允许在一个集群中配置的 Bucket 在另一个集群中可用。这仅在对象存储后端本身可以从两个集群访问时才有效。
+
+跨提供商可移植性是指允许组织或团队无缝地从一个对象存储提供商迁移到另一个对象存储提供商，
+而无需更改应用程序定义（PodTemplates、StatefulSets、部署等）。这只有在源和目标提供者使用相同的数据时才有可能。
```
Comment (zhangxyjlu) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:178]: Deployment 不需要翻译

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+
+Stay tuned for what comes next, and if you have any questions, comments or suggestions
+
+* Chat with us on the Kubernetes [Slack:#sig-storage-cosi](https://kubernetes.slack.com/archives/C017EGC1C6N)
+* Join our [Zoom meeting](https://zoom.us/j/614261834?pwd=Sk1USmtjR2t0MUdjTGVZeVVEV1BPQT09), every Thursday at 10:00 Pacific Time
+* Participate in the [bucket API proposal PR](https://github.com/kubernetes/enhancements/pull/2813) to add your ideas, suggestions and more.
+-->
+## 接下来
+
+令人惊叹的 sig-storage-cosi 社区一直在努力将 COSI 标准带入 alpha 状态。
```
Comment (zhangxyjlu) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:197]:
```suggestion
令人惊叹的 sig-storage-cosi 社区一直在努力将 COSI 标准带入 Alpha 状态。
```

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 13]
```diff
@@ -0,0 +1,208 @@
...
+## API
+
+The COSI API is centered around buckets, since bucket is the unit abstraction for object storage. COSI defines three Kubernetes APIs aimed at managing them
+
+* Bucket
+* BucketClass
+* BucketClaim
+-->
+## 接口
+COSI 接口 以 Bucket 为中心，因为 Bucket 是对象存储的抽象单元。COSI 定义了三个旨在管理它们的 Kubernetes API
```
Comment (howieyuen) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:106]: 标题后需空行

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 14]
```diff
@@ -0,0 +1,208 @@
...
+## Self-Service
+
+Other than providing kubernetes-API driven bucket management, COSI also aims to empower DevOps personnel to provision and manage buckets on their own, without admin intervention. This, further enabling dev teams to realize faster turn-around times and faster time-to-market. 
+
+COSI achieves this by dividing bucket provisioning steps among two different stakeholders, namely the administrator (admin), and the cluster operator. The administrator will be responsible for setting broad policies and limits on how buckets are provisioned, and how access is obtained for them. The cluster operator will be free to create and utilize buckets within the limits set by the admin. 
+
+For example, a cluster operator could use an admin policy could be used to restrict maximum provisioned capacity to 100GB, and developers would be allowed to create buckets and store data upto that limit. Similarly for access credentials, admins would be able to restrict who can access which buckets, and developers would be able to access all the buckets available to them.
+-->
+## 自助服务
+除了提供 kubernetes-API 驱动的 Bucket 管理之外，COSI 还旨在使 DevOps 人员能够自行配置和管理 Bucket，
```
Comment (howieyuen) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:145]: 标题后需空行

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 15]
```diff
@@ -0,0 +1,208 @@
...
+而无需管理员干预。 这进一步使开发团队能够实现更快的周转时间和更快的上市时间。
```
Comment (howieyuen) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:146]:
```suggestion
而无需管理员干预。这进一步使开发团队能够实现周转时间更短，上市更快。
```

**Review by howieyuen — CHANGES_REQUESTED** [review]

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md (hunk 6 above)
Comment (my-git9) [reply content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:37]: 存储一般是不可缩容的

File: content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md [hunk 16]
```diff
@@ -0,0 +1,210 @@
...
+layout: blog
+title: "Introducing COSI: Object Storage Management using Kubernetes APIs"
+date: 2022-09-02
+slug: cosi-kubernetes-object-storage-management
+-->
+
+<!--
+**Authors:** Sidhartha Mani ([Minio, Inc](https://min.io))
+-->
+**作者:** Sidhartha Mani ([Minio, Inc](https://min.io))
```
Comment (Sea-n) [inline content/zh-cn/blog/_posts/2022-09-02-cosi-kubernetes-object-storage-management.md:18]:
```suggestion
**作者：**Sidhartha Mani ([Minio, Inc](https://min.io))
```

**Review by Sea-n — COMMENTED** [review]
/approve

**Pull request merged**